src/Microsoft.AspNetCore.Mvc.Core/ProducesResponseTypeAttribute.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.Formatters;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    /// <summary>
+    /// Specifies the type of the value and status code returned by the action.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = true, Inherited = true)]
+    public class ProducesResponseTypeAttribute : Attribute, IApiResponseMetadataProvider, IFilterMetadata
+    {
+        /// <summary>
+        /// Initializes an instance of <see cref="ProducesResponseTypeAttribute"/>.
+        /// </summary>
+        /// <param name="type">The <see cref="Type"/> of object that is going to be written in the response.</param>
+        /// <param name="statusCode">Http response status code</param>
+        public ProducesResponseTypeAttribute(Type type, int statusCode)
+        {
+            if (type == null)
+            {
+                throw new ArgumentNullException(nameof(type));
+            }
+
+            Type = type;
+            StatusCode = statusCode;
+        }
+
+        /// <summary>
+        /// The type of the value returned by an action.
+        /// </summary>
+        public Type Type { get; set; }
+
+        /// <summary>
+        /// HTTP status code of the response.
+        /// </summary>
+        public int StatusCode { get; set; }
+
+        /// <inheritdoc />
+        public void SetContentTypes(MediaTypeCollection contentTypes)
```

Comment:
Why do we need to implement this at all? I understand that it causes no harm, but it's confusing that it has this here.